crates/evm/execution-types/src/chain.rs

```diff
@@ -538,19 +538,50 @@ pub(super) mod serde_bincode_compat {
     /// ```
     #[derive(Debug, Serialize, Deserialize)]
     pub struct Chain<'a> {
-        blocks: BTreeMap<BlockNumber, SealedBlockWithSenders<'a>>,
+        blocks: SealedBlocksWithSenders<'a>,
```

Comment:
I'm having quite a few issues dealing with the lifetimes here:
- This won't compile because of the lifetime requirement for `Deserialize`, which I can fix if I add `#[serde(borrow)]`.
- Adding the `#[serde(borrow)]` attributes transfers the problem down to the `ExExNotification` enum, which now also requires the same attribute.
- Compilation finally fails on because `DeserializedOwned` isn't implemented due to the above lifetimes limitations.

Wanted to check with you if this was the correct path and I need to fix some lifetimes issues or if I am attacking the problem incorrectly.